src/ValidatingHydrator.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Input\Validation;
+
+use Yiisoft\Hydrator\HydratorInterface;
+use Yiisoft\Input\Validation\Attribute\EarlyValidationResolver;
+use Yiisoft\Validator\Result;
+use Yiisoft\Validator\ValidatorInterface;
+
+final class ValidatingHydrator implements HydratorInterface
+{
+    public function __construct(
+        private readonly HydratorInterface $hydrator,
```

Comment:
`readonly` is available starts from php 8.1

Reply:
fixed